### DISCUSSION TITLE
R-Instat and Text Mining

### DISCUSSION THREAD

**Discussion by rdstern** [IDEMSInternational/R-Instat#6695] [General]
We have so rar discussed 4 books.  A fifth one is Called [Text Mining with R: A Tidy Approach](https://www.tidytextmining.com/index.html)

Like the others, it is available in print, but can also be read freely on the web.  And, like the others, it assumes proficiency in R as a start.

[Text mining with R-Instat.docx](https://github.com/africanmathsinitiative/R-Instat/files/7032416/Text.mining.with.R-Instat.docx)

The attached document is my first go at taking part of a book and examining how well R-Instat can be used in following the materials.  Through this, I have found various places where dialogues could do with improvements.  This could now be used by the documentation team and possibly improved steadily as facilities are added to R-Instat.  It can also motivate improvements to some of the dialogues.

**Comment by Vitalis95** [comment]
@rdstern I have been going through the text mining document which you shared with me. I would like to inquire how this should implemented(as a new dialog?). I have understood how this works and below is my summary of what I understood from document.

Can we have Text mining as a dialog in R-Instat
Where the text data is turned into tidy text dataset using the unnest tokens function. We can also examine how often positive and negative words (sentiment lexicon) occurred in the tidy text using inner_join and get_sentiments functions.   We can extract insights from sentiment lexicon and tidy text by summarizing using statistical techniques, visualizing and also we my fit topical models like LDA to understand how words are associated with topics.

I am having a challenge in installing some of the packages in R-Instat. Just to mention a few of them; 
•	tidytext 
•	janeaustenr

I am hoping to hear from you soon. Thank you.

**Comment by rdstern** [comment]
I am not sure you can have followed the text sufficiently.  We already have janeaustenr in R-Instat (at least I do and I don't think I added it specially) If not, then you should anyway be able to add a new package using the Tools menu.  The dialogue will also tell you if it is already installed.
I am not convinced you need a general text mining dialogue, but we do need the unnest_tokens to be within a dialogue.  That is ambitious for you to add, but it could become one of your tasks for the future.  In that task you should also contact David in the sprint to check that the concept is ok of where that dialogue - or button should be placed.  (You can always message him on github, but he ignores that, so a skype message is needed for him.)
At a more modest level the text you examined showed the need to improve the Describe > One Variable > Frequencies dialogue and that is a good task for you to work on soon.
